packages/playwright-core/src/server/injected/roleUtils.ts

```diff
@@ -1005,13 +1005,18 @@ function isNativelyDisabled(element: Element) {
   return isNativeFormControl && (element.hasAttribute('disabled') || belongsToDisabledFieldSet(element));
 }
 
-function belongsToDisabledFieldSet(element: Element | null): boolean {
-  if (!element)
+function belongsToDisabledFieldSet(element: Element): boolean {
+  const fieldSetElement = element?.closest('FIELDSET');
+  if (!fieldSetElement)
+    return false;
+  if (!fieldSetElement.hasAttribute('disabled'))
     return false;
```

Comment:
This does not account for `<fieldset disabled><fieldset><input>`, where input should be "disabled" due to grandparent fieldset.